chore(ios): updated GoogleMaps SDK and Google-Map-Utils

pinpong opened the pull request:
Changelog see: https://developers.google.com/maps/documentation/ios-sdk/release-notes
Changelog see: https://github.com/googlemaps/google-maps-ios-utils/releases

fix: https://github.com/react-native-maps/react-native-maps/issues/5228
fix: https://github.com/react-native-maps/react-native-maps/issues/5250

Breaking changes: Raise the minimum supported iOS version to 15.0

@salah-ghanim tested on my side, please merge.

pinpong commented:
@salah-ghanim i've update the react native to 0.77.1 within this PR, hope this is ok.

salah-ghanim commented:
@pinpong thank you so much for the contribution 🔥 

can you please share a full / detailed description of your changes? I see a large diff and would really like not to have to review it without context.

pinpong commented:
Updated the example to RN 0.77.1
Migrated ios example to swift
Updated min iOS version for Apple maps to 13
Updated Google maps iOS sdk
Updated package dependencies and minimum require versions

basitmir commented:
@salah-ghanim can you please review this. This will solve all the problems w.r.t new version of react-native

salah-ghanim commented:
@basitmir I wonder what do you mean by it will solve all problems with react-native new versions

salah-ghanim reviewed (approved):
LGTM
